File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    for i in range(1, n + 1):
        target = i
        temp = 0
        while (temp <= n):
            if temp == n:
                answer += 1
                break
                
            temp += target
            target += 1
            
    return answer

File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    for (int i = 1; i <= n; i++) {
        int target = i;
        int temp = 0;
        while (temp <= n) {
            if (temp == n) {
                answer++;
                break;
            }
            
            temp += target;
            target++;
        }
    }
    
    return answer;
}